sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/models/KeyValueFields.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.data.appconfiguration.implementation.models;
+
+import com.azure.core.util.ExpandableStringEnum;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import java.util.Collection;
+
+/** Defines values for KeyValueFields. */
+public final class KeyValueFields extends ExpandableStringEnum<KeyValueFields> {
```

Comment:
Let's add a Swagger transform and code customization to have this class be renamed to `SettingsField`, rename `LOCKED` to `IS_READ_ONLY`, and add the `toStringMapper` API and then put this class into the public API. With that we no longer have to do the `SettingFields` to `KeyValueFields` mapping, and vice versa.

Reply:
Created an issue to address this https://github.com/Azure/azure-sdk-for-java/issues/33993